checkbox-provider-ce-oem/bin/cpufreq_governors.py

```diff
@@ -0,0 +1,850 @@
+        --powersave: Runs the Powersave Governor test.
+        --ondemand: Runs the Ondemand Governor test.
+        --conservative: Runs the Conservative Governor test.
+        --schedutil: Runs the Schedutil Governor test.
+
+    Returns:
+        int: The exit code of the test run. 0 if successful, 1 otherwise.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-q", "--quiet", action="store_true", help="Suppress output."
+    )
+    parser.add_argument(
+        "-c",
+        "--capabilities",
+        action="store_true",
+        help="Only output CPU capabilities.",
+    )
+    parser.add_argument(
+        "-d",
+        "--debug",
+        action="store_true",
+        help="Turn on debug level output for extra info during test run.",
+    )
+    parser.add_argument(
+        "--resource-format",
+        action="store_true",
+        help="Print the capabilities in Checkbox resource job format.",
+    )
+    parser.add_argument(
+        "--userspace",
+        action="store_true",
+        help="Run Userspace Governor test",
+    )
+    parser.add_argument(
+        "--performance",
+        action="store_true",
+        help="Run Performance Governor test",
+    )
+    parser.add_argument(
+        "--powersave",
+        action="store_true",
+        help="Run Powersave Governor test",
+    )
+    parser.add_argument(
+        "--ondemand",
+        action="store_true",
+        help="Run Ondemand Governor test",
+    )
+    parser.add_argument(
+        "--conservative",
+        action="store_true",
+        help="Run Conservative Governor test",
+    )
+    parser.add_argument(
+        "--schedutil",
+        action="store_true",
+        help="Run Schedutil Governor test",
+    )
+    args = parser.parse_args()
+
+    logger = init_logger()
+
+    if args.debug:
+        logger.setLevel(logging.DEBUG)
+
+    if args.quiet or args.resource_format:
+        # Not logging anything
+        logger.setLevel(logging.CRITICAL + 1)
+
+    test = CPUScalingTest()
+    if not test.get_cpu_freq_directories():
+        logging.info("CPU Frequency Scaling not supported")
+        return 1
+
+    if args.resource_format:
+        logging.getLogger().setLevel(logging.ERROR)
+        return 0 if test.get_supported_governors(resource_format=True) else 1
+
+    if not test.get_system_capabilities():
+        logging.error("Failed to get system capabilities")
+        return 1
+
+    exit_code = 0
+
+    test_methods = {
+        "userspace": test.test_userspace,
+        "performance": test.test_performance,
+        "powersave": test.test_powersave,
+        "ondemand": test.test_ondemand,
+        "conservative": test.test_conservative,
+        "schedutil": test.test_schedutil,
+    }
+
+    for arg, test_method in test_methods.items():
+        if getattr(args, arg):
+            if not test_method():
+                exit_code = 1
+
+    test.restore_governors()
```

Comment:
We are going to run one specific test at one time, right?
If so, I would suggest that we uses getattr to get the test function and call it, instead of checking the method in for loop. Likes
```
try:
    if not getattr(test, f"test_{args.tests}")():
        exit_code = 1
except:
    print("not supported test method")
```

Reply:
Thanks for the suggestion, I'll apply it to the next commit.